.github/workflows/_test_containerized.yml

```diff
@@ -28,15 +29,29 @@ jobs:
           - go-version: '1.22'
             image: golang:{0}-buster
           # The amazonlinux:2 variant is only relevant for the default go version yum ships (currently 1.20)
-          - go-version: '1.19'
-            image: amazonlinux:2
           - go-version: '1.21'
             image: amazonlinux:2
           - go-version: '1.22'
             image: amazonlinux:2
-    name: ${{ toJSON(matrix) }}
-    runs-on: ubuntu-latest
+    name: ${{ matrix.arch }} ${{ format(matrix.image, matrix.go-version) }} go${{ matrix.go-version }}${{ matrix.waf-log-level && format(' (DD_APPSEC_WAF_LOG_LEVEL={0})', matrix.waf-log-level) || '' }}
+    # We use ARM runners when needed to avoid the performance hit of QEMU
+    runs-on: ${{ matrix.arch == 'amd64' && 'ubuntu-latest' || 'arm-4core-linux' }}
     steps:
+      # Docker is not present on early-access ARM runners
+      - name: Prepare ARM Runner
+        if: matrix.arch == 'arm64'
+        run: |-
+          for pkg in docker.io docker-doc docker-compose docker-compose-v2 podman-docker containerd runc; do sudo apt-get remove -y $pkg || echo "Not present: $pkg"; done
```

Comment:
Isn't there an apt action ? (that would support caching for us)

Reply:
The early-access ARM runners don't have a full fledged image and lack a lot of tools needed by some actions, so even if there was one (I don't know about one), it's not certain it'd work there...